shared-libs/tombstone-utils/src/tombstone-utils.js

```diff
@@ -29,11 +37,48 @@ var saveTombstone = function(DB, doc, logger) {
 };
 
 var getDoc = function(Promise, DB, change) {
-  if (change.doc) {
+  if (change.doc && !isDeleteStub(change.doc)) {
     return Promise.resolve(change.doc);
   }
 
-  return DB.get(change.id, { rev: change.changes[0].rev });
+  return DB
+    .get(change.id, { rev: change.changes[0].rev, revs: true })
+    .then(function(doc) {
+      if (!isDeleteStub(doc)) {
+        return doc;
+      }
+
+      // we've received a delete stub doc
+      var previousRevision = getPreviousRev(doc._revisions);
+      if (!previousRevision) {
+        return doc;
+      }
+
+      return DB.get(change.id, { rev: previousRevision });
+    });
+};
+
+// CouchDB/Fauxton deletes don't include doc fields in the deleted revision
+var isDeleteStub = function(doc) {
+  // determines if array2 is included in array1
+  var arrayIncludes = function(array1, array2) {
+    return array2.every(function(elem) {
+      return array1.indexOf(elem) !== -1;
+    });
+  };
+
+  return arrayIncludes(COUCHDB_TOMBSTONE_PROPERTIES, Object.keys(doc)) && !!doc._deleted;
+};
+
+// Returns n-th previous rev
+// @param {Object} revisions - doc _revisions
+// @param {number} n
+var getPreviousRev = function(revisions, n) {
+  n = n || 1;
```

Comment:
It looks like n is never given. Can we hardcode this to 1 and remove the parameter for simplicity?

Reply:
Done.